File: core/documents/argument_graph_engine.py
from __future__ import annotations

from typing import Any, Dict, List

from core.documents.rhetorical_structure_engine import extract_rhetorical_structure


def build_argument_graph(text: str) -> Dict[str, Any]:
    rhet = extract_rhetorical_structure(text)
    headings = [u for u in rhet["units"] if u.get("type") == "heading"]
    nodes = [{"id": f"h{i}", "role": "claim" if i == 0 else "support", "title": h.get("title", "")} for i, h in enumerate(headings)]
    edges = [{"from": nodes[i]["id"], "to": nodes[i + 1]["id"]} for i in range(len(nodes) - 1)]
    return {"nodes": nodes, "edges": edges, "deterministic_inputs": [f"nodes={len(nodes)}"]}

File: src/documents/argumentGraphEngine.ts
/**
 * Converted from Python: core/documents/argument_graph_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { extractRhetoricalStructure } from "./rhetoricalStructureEngine.js";

export function buildArgumentGraph(text: any): any {
  var rhet: any = extractRhetoricalStructure(text);
  var headings: any = py.iter(py.at(rhet, "units")).filter((u: any) => py.eq(py.get(u, "type"), "heading")).map((u: any) => u);
  var nodes: any = py.enumerate(headings).map(([i, h]: any) => ({"id": `h${py.toStr(i)}`, "role": (py.eq(i, 0) ? "claim" : "support"), "title": py.get(h, "title", "")}));
  var edges: any = py.range(py.sub(py.len(nodes), 1)).map((i: any) => ({"from": py.at(py.at(nodes, i), "id"), "to": py.at(py.at(nodes, py.add(i, 1)), "id")}));
  return {"nodes": nodes, "edges": edges, "deterministic_inputs": [`nodes=${py.toStr(py.len(nodes))}`]};
}
export { extractRhetoricalStructure };
